src/plugins/factoids/factoidsPlugin.js

```diff
@@ -1,35 +1,66 @@
-const facts = require('./facts.json').factoids;
 const fs = require('fs');
+const { promisify } = require('util');
+const { getStore } = require('./storage.persistent');
+
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+
+async function checkAndSaveDisabled(msg) {
+  const safeTo = String(msg.channel)
+    .replace(/[^a-zA-Z0-9#]/g, (char) => '-' + char.codePointAt(0).toString(16) + '-')
+    .replace(/#/g, '-H')
+    .replace(/^([^-])/, '-$1');
+  const disableFile = `/tmp/disable-factoids${safeTo}`;
+
+  const disabled = await readFile(disableFile).then(() => true, () => false);
+
+  if (!disabled && msg.from === 'ecmabot') {
+    await writeFile(disableFile, 'delete to enable', () => {});
+  }
+
+  if (disabled) {
+    console.log(`Factoids are disabled in ${msg.channel} due to an ecmabot message.`);
+    console.log(`    Delete ${disableFile} to reset this.`);
+  }
+
+  return disabled;
+}
 
 const factoidPlugin = async (msg) => {
-  try {
-    await fs.readFileSync('/tmp/disable-factoids');
+  if (!msg.command) return null;
+  if (await checkAndSaveDisabled(msg)) {
     return;
-  } catch (e) {}
-  if (msg.from === 'ecmabot') {
-    fs.writeFile('/tmp/disable-factoids', 'x', () => {});
   }
-  if (!msg.command) return null;
 
-  let parts = msg.command.command.split('@').map((x) => x.trim());
-  const [key, nick] = parts;
+  const { command } = msg.command;
 
-  const fact = facts[key];
-  if (!fact) return;
+  const STORE = getStore();
+
+  if (STORE.needsLoadFromDisk()) {
+    await STORE.loadFromDisk();
+  }
 
-  let value = fact.value;
-  if (fact.alias) {
-    const fact2 = facts[fact.alias];
-    if (fact2) {
-      value = fact2.value;
+  const learnMatch = command.match(
+    /learn\s+([a-zA-Z0-9_ -]{2,}[a-zA-Z0-9_-])\s*=\s*(.*)$/,
```

Comment:
Please sanity check this.

Reply:
what is it supposed to match and not match, exactly?

Reply:
That's a good question, and I'm not sure what the answer is. We currently actually allow a lot of symbols, so I should probably make it more general, like `!learn <ascii...> = anything, but maybe not zalgo`